### PR TITLE
arch/xtensa: Remove obsolete esp32 spi-flash driver.

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
Remove obsolete esp32 spi-flash driver files,
latest one is already moved to directory external/esp_idf_port.
@sunghan-chang 
@xixidodo

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [93f586d9d1f9fc5a32e75a849532d84a69b882db] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [93f586d9d1f9fc5a32e75a849532d84a69b882db] - Code Rule Check OK.

File: os/arch/xtensa/src/esp32/Make.defs [hunk 1]
```diff
@@ -124,13 +124,6 @@ ifeq ($(CONFIG_I2C),y)
 CHIP_CSRCS += esp32_i2c.c
 endif
 
-ifeq ($(CONFIG_FS_SMARTFS),y)
-CHIP_CSRCS += esp32_flash.c
```
Comment (SRCN-LiQin) [inline os/arch/xtensa/src/esp32/Make.defs]: spiflash source file is built under the config of SMARTFS, while in esp_idf_port, it's depend on the WIFI support config, is it clear dependency for both external and os/arch  side ?

Comment (ChinaTizenRT) [reply os/arch/xtensa/src/esp32/Make.defs]: this PR is to just remove legacy spi-flash driver, actually it does not used now, althou compiled.
then spi flash driver just only depends  WIFI config, not both.
it is better that make new PR to seperate spi flash driver from wifi support, i think.

**Comment by sunghan-chang** [comment]
@ChinaTizenRT @SRCN-LiQin Could you let me know the conclusion of dependency?
In my opinion, `seperate spi flash driver from wifi support.` is necessary.

**Comment by seinfra** [comment]
Target : [ae5e06725867686d24fa6e70ab29c0b4744854e8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ae5e06725867686d24fa6e70ab29c0b4744854e8] - Code Rule Check (C++) OK.

File: os/arch/xtensa/src/esp32/Make.defs (hunk 1 above)
Comment (ChinaTizenRT) [reply os/arch/xtensa/src/esp32/Make.defs]: In my opinion, seperate spi flash driver from wifi support. is necessary.
  -- okay,  alreay seperate spi flash driver from wifi support, that is move spi flash driver 
to os/arch/xtensa/chip directory.

**Comment by seinfra** [comment]
Target : [52d9ac11b1aa0d37d3f4d6a65c484efce4f3b9c3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [52d9ac11b1aa0d37d3f4d6a65c484efce4f3b9c3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [60d50697149089255b2f9dab1db24bf8f1a22c7a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [60d50697149089255b2f9dab1db24bf8f1a22c7a] - Code Rule Check (C++) OK.

**Pull request merged**